onnx/shape_inference/implementation.h

```diff
@@ -12,6 +12,39 @@
 namespace ONNX_NAMESPACE {
 namespace shape_inference {
 
+struct SymbolicShape {
+  SymbolicShape() : index_(0){}
+  std::string createNew() {
+    return symbol_prefix + std::to_string(++index_);
+  };
+  void init(GraphProto* g) {
+    index_ = 0;
+    checkExistingSymbolicShape(*g->mutable_input());
+    checkExistingSymbolicShape(*g->mutable_output());
+    checkExistingSymbolicShape(*g->mutable_value_info());
+  }
+  private:
+    unsigned int index_;
+    const std::string symbol_prefix = "unk_";
+    template<class T>
+    void checkExistingSymbolicShape(T protos) {
+      for (auto& proto : protos) {
+        auto tensorType = proto.mutable_type()->mutable_tensor_type();
+        if (tensorType->has_shape()) {
+          for (int j = 0; j < tensorType->shape().dim_size(); ++j) {
+            if (tensorType->shape().dim(j).has_dim_param()) {
+              unsigned int existingIndex = 0;
+              sscanf(tensorType->shape().dim(j).dim_param().c_str(), (symbol_prefix + "%d").c_str(), &existingIndex);
```

Comment:
Although using sscanf here can easily detect "unk_XXX", sscanf might be a dangerous function...

Reply:
I dont exactly get whats happening here... Are you assuming the model will have symbols in a specific numerical range ? Why no simply maintain a list of all symbols in the graph and then when you create a new one just make sure it doesnt already exist

Reply:
My intention was to save some memory (additional existing shape hashset) and time... For instance, if "unk_1", "unk_2"..."unk_50"  are used, using a list to check will go through "unk_1" to "unk_50" and then finally find "unk_51" is usable. This will take more time... but as you said, to assume "unk_XXX" here, the logic will look quite confusing. I am changing the logic to maintain a shape list and check. Thanks for the review!

Reply:
well the time depends on the data structure you use.